### PR TITLE
Fake Input and Output streams to to test CLI class

### PR DESCRIPTION
**Pull request by sarahabimay**
@maikon @jsuchy

### CODE REVIEW

**Comment by octopusinvitro** [comment]
Everything is ok from my ignorant point of view.
:+1: 
:octopus:

File: src/CLI.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    private BufferedReader readStream;
+    private PrintStream writeStream;
+    private Game game;
+    private Symbol consoleSymbol;
+
+    public CLI(InputStream inputStream, PrintStream outputStream, Game game) {
+        this.readStream = new BufferedReader(new InputStreamReader(inputStream));
+        this.writeStream = outputStream;
+        this.game = game;
+        Symbol.setOrdinalToSymbol();
```
Comment (jsuchy) [inline src/CLI.java:18]: Static method smells here, I haven't gotten to what this does yet, but it's smelly.

File: src/CLI.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    }
+
+    public String displayGreeting() {
+        String greeting = GAME_GREETING;
+        writeStream.println(greeting);
+        return greeting;
```
Comment (jsuchy) [inline src/CLI.java]: Why return the greeting? displayGreeting is a command, I'd expect it to be void and just have the side effect.

File: src/CLI.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    }
+
+    public Symbol requestConsoleMove() {
+        while (consoleSymbol == null) {
+            writeStream.println(CONSOLE_MOVE_REQUEST);
+            consoleSymbol = Symbol.getSymbolFromOrdinal(readInput());
+        }
+        return consoleSymbol;
```
Comment (jsuchy) [inline src/CLI.java:32]: Shouldn't consoleSymbol just be a local? I wouldn't expect this to be a field as we now have state to worry about.

File: src/CLI.java [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    public void displayAIMove(Symbol symbol) {
+        writeStream.println(String.format(AI_MOVE_DISPLAY, symbol));
+    }
+
+    public void play() {
+        displayGreeting();
+        displayConsoleMove(requestConsoleMove());
+        playAgainstAI();
+        displayResult();
+        playAgain();
```
Comment (jsuchy) [inline src/CLI.java]: Avoid recursion here. https://blog.8thlight.com/doug-bradbury/2015/04/27/stack-overflow.html

File: src/CLI.java [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+
+    private int readInput() {
+        try {
+            return Integer.parseInt(readStream.readLine());
+        } catch (IOException e) {
+            e.printStackTrace();
+        } catch (NumberFormatException e) {
+            return 0;
+        }
+        return 0;
```
Comment (jsuchy) [inline src/CLI.java:86]: There duplication here, how can we remove it?

File: src/Game.java [hunk 6]
```diff
@@ -3,46 +3,47 @@
...
+        this.rules = rules;
+        Symbol.setOrdinalToSymbol();
     }
 
-    public SYMBOL generateAIMove() {
-        int range = SYMBOL.ordinalToSymbol.size();
+    public Symbol generateAIMove() {
+        int range = Symbol.ordinalToSymbol.size();
         aiPlayer = getRandomSymbolInRange(range);
         return aiPlayer;
```
Comment (jsuchy) [inline src/Game.java:18]: Is this field used anywhere else? Again, avoid fields if possible, especially if they can be reassigned.

File: src/CLI.java [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+    public CLI(InputStream inputStream, PrintStream outputStream, Game game) {
+        this.readStream = new BufferedReader(new InputStreamReader(inputStream));
+        this.writeStream = outputStream;
+        this.game = game;
+        Symbol.setOrdinalToSymbol();
+    }
+
+    public String displayGreeting() {
+        String greeting = GAME_GREETING;
+        writeStream.println(greeting);
```
Comment (gemcfadyen) [inline src/CLI.java]: Why not inline the greeting variable here and just use the constant?

File: src/CLI.java (hunk 1 above)
Comment (sarahabimay) [inline src/CLI.java:18]: That was Makis's solution and suggestion!

File: src/CLI.java (hunk 2 above)
Comment (sarahabimay) [inline src/CLI.java]: Agreed. Was originally for testing but no need now.

File: src/CLI.java (hunk 4 above)
Comment (sarahabimay) [inline src/CLI.java]: I do this in TTT too.

File: src/CLI.java (hunk 5 above)
Comment (sarahabimay) [inline src/CLI.java:86]: which duplication?  the return 0?

**Pull request merged**